File: ClimbingStairs/ClimbingStairs.py
class Solution:
    def climbStairs(self, n: int) -> int:

        result_list = []
        result_list.append([n])
        result_cnt = 0

        for element in range(n):
            temp_list = []
            for index in range(len(result_list[element])):

                value_1 = result_list[element][index] - 1
                value_2 = result_list[element][index] - 2
                
                if value_2 == 0:
                    result_cnt = result_cnt + 1
                if value_1 == 0:
                    result_cnt = result_cnt + 1
                if value_2 > 0:
                   temp_list.append(value_2)
                   value_2 = 0
                else:
                    value_2 = 0

                if value_1 > 0:
                    temp_list.append(value_1)
                    value_1 = 0
                else:
                    value_1 = 0

            result_list.append(temp_list)
        return result_cnt


if __name__ == '__main__':
    SolutionObj = Solution()
    print(SolutionObj.climbStairs(45))

File: ClimbingStairs/ClimbingStairs.cpp
#include <iostream>
#include <unordered_map>

class Solution {
public:
    int climbStairs(int n) {

        std::unordered_map<int, int> results;

        return aux_climbingStairs(n, results);
    }

    int aux_climbingStairs(int n, std::unordered_map<int, int>& map)
    {
        if ((n == 1) || (n == 2))
        {
            return n;
        }

        if (map.find(n) == map.end())
        {
            map[n] = aux_climbingStairs(n - 1, map) + aux_climbingStairs(n - 2, map);
        }
        
        return map[n];
    }
};

int main()
{
    int n = 3;

    Solution SolutionObj;
    std::cout << SolutionObj.climbStairs(n);
    return 0;
}